### PR TITLE
Upgraded for 4.x

### PR DESCRIPTION
**Pull request by stdavis**
Basic functionality. No LOD stuff or tests yet.

### CODE REVIEW

File: tests/example-happy-path-tokens.html [hunk 1]
```diff
@@ -28,31 +28,35 @@
...
+            'esri/views/MapView',
             'esri/geometry/Extent',
 
             'dojo/dom-construct',
 
             'dojo/domReady!'
-        ], function(Module, Map, Extent, domConstruct) {
-            var quadWord = '';
+        ], function (Module, Map, MapView, Extent, domConstruct) {
+            var quadWord = 'alabama-anvil-picnic-sunset';
```
Comment (steveoh) [inline tests/example-happy-path-tokens.html]: this should be scrubbed.

File: LayerSelector.js [hunk 2]
```diff
@@ -367,14 +349,15 @@ define([
 
                     resolvedInfos.push({
                         Factory: WebTiledLayer,
-                        url: layer.urlPattern.replace('{quad}', this.quadWord),
+                        urlTemplate: layer.urlPattern.replace('{quad}', this.quadWord),
                         linked: linked,
                         id: id,
-                        tileInfo: tileInfo,
+                        // tileInfo: tileInfo,
```
Comment (steveoh) [inline LayerSelector.js]: tileinfo is unused now. should probably delete for eslint.

File: tests/example-happy-path-tokens.html (hunk 1 above)
Comment (stdavis) [inline tests/example-happy-path-tokens.html]: done

File: LayerSelector.js (hunk 2 above)
Comment (stdavis) [inline LayerSelector.js]: done

**Comment by stdavis** [comment]
I believe that this is ready for review. 

Travis-ci.com isn't wired up yet. I've submitted the request for this repo to be migrated but they haven't been as quick to respond.

**Pull request merged**